### DISCUSSION TITLE
Redesign navigation view

### DISCUSSION THREAD

**Discussion by TobiPeterG** [osmandapp/OsmAnd#20925] [Ideas]
OSMand shows all information I wish related to navigating.
However, they often look outdated or have some kind of issues (at least for me).

For example, in my case, usability issues are:
A) long street names are cut off
B) the lane view is too small
C) no obvious way to stop the navigation (making the "back" gesture just closes the app)
D) Elements are in general quite small
E) warnings about upcoming events appear only shortly before them. Having a banner properly warning of the event with a bar showing how far it is to the event would be great instead of "just" the icon. Maybe highlight them on the map as well?

Design ideas:
A) Have a bar at the bottom showing the time and distance remaining to the next stop/final stop. A bar showing the progress, maybe highlighting POIs (gas stations, parking spots, radar cams) would be awesome
This bar could be swiped up to show a "stop" button, a button to toggle voice announcements, a button to show an overview of the complete route, a button to re-calculate the route from the current position and a settings button
This could also include common POIs like gas stations etc. To route to on the way

B) the "current speed" and "speed limit" signs so not fit to each other, they have a very different look. It would be great if the current speed could be shown in a white circle, maybe connected through a rounded rectangle with the current speed limit?

C) have a top bar that has an indicator of the next action on the left (so right/left arrow) with the distance below or next to it on the right and the road one will go to with that action. Underneath, show the action after that with a small text "after that" and an icon maybe? The current street could be shown there as well, but it would be difficult to not confuse the current street with the next street. So maybe show a little text underneath the current position in a floating element with the current street name?
Or have a separator line, so the icon of the current street is in the top left and the street name could be next to it filling up to two rows? Or show the current street with the icon of it in the bottom bar at the very bottom?

D) show the street names of the roads one is going to drive on. The names might not always fit on the road itself, maybe have a floating indicator like a speech bubble showing the name

E) show the route overview before starting a route in 2D mode and not in 3D mode, even when 3D mode is selected, as it's hard to see the actual route in that mode

F) Increase the size of the lanes one has to take. Currently this is very small. Maybe the display could slide in from the top above the top bar (where the next action would be shown below it) when it's relevant, using the horizontal space to clearly show it?)
It would also be nice if a lane display would always also mean that an action is displayed for that segment, but that is a routing thing.

Of course, not all of this needs to be implemented, I'd also be happy if this creates some discussions.
The data is already there, so it's mostly a design task, but I also see that OSMand currently has the ability to select where which widget is shown, but maybe that isn't that useful?
Where do different users have the widgets?
Maybe it would be enough to allow adding/removing buttons from the bottom bar and moving elements from the bottom bar to the top bar and vice-versa? Would that be even used?

Should other users have additional ideas, feel free to add it :)
I would also be very happy to help create design elements like the bars and align them etc :)

**Comment by ZeljkoBG** [comment]
NOTE!: every remark of mine is well-intentioned

My remarks, global

1.
@TobiPeterG what you call 'look outdated' I call "Picture book", is design entire former 'eastern bloc'. 90% applications recognize whether from the east or west Europe.

Example:
![widget_context_menu_andr-721d0d58f3f2024298c59ebc7c3a1e1a](https://github.com/user-attachments/assets/5a627877-aa00-4906-b527-89147e788c3a)

See: Yandex; Navitel, GeoNET, Navi-maps, bGEO, Automapa, MapaMaps, Navmii, City Guide, Navikey,...  and OsmAnd design as if they were designed by the same man.
Hardly anything will change there.

2.
What can be changed, and they don't think they should, is 'scattering of dialog boxes' per application. There is no order, no logic. OsmAnd tuning requires a good faculty.
'Local settings' (Voice, app language, map language, measurement units,...) in well-designed GPS applications, are in one place. One dialog box.

3.
The application is too bulky. As someone wrote, 'OsmAnd is not a GPS app but a GPS tool'.
